### PR TITLE
Show names of missing expansion packs; cleanup

### PR DESCRIPTION
**Pull request by Gymnasiast**
Since OpenRCT2 uses an RCT2.EXE that lacks information about the expansion packs, attempting to load a park using an object from one of them will show a generic message saying it requires an add-on pack, rather than telling the user which one. This is fixed by this PR. Also took the oppertunity to do some cleanup in the function itself.

### CODE REVIEW

File: src/object_list.c [hunk 1]
```diff
@@ -501,17 +505,25 @@ void set_load_objects_fail_reason(){
 		return;
 	}
 
-	char* exapansion_name = &RCT2_ADDRESS(RCT2_ADDRESS_EXPANSION_NAMES, char)[128 * expansion];
-	if (*exapansion_name == '\0'){
+	switch(expansion) {
+		case 1:
+			expansion_name="Wacky Worlds";
```
Comment (IntelOrca) [inline src/object_list.c]: Should this not be localised?

Comment (Gymnasiast) [inline src/object_list.c]: Good point. I'll change that.

**Comment by Gymnasiast** [comment]
Ok, it uses localised names for the expansions now.

**Comment by IntelOrca** [comment]
Good, now I will suggest that you edit STR_REQUIRES_THE_FOLLOWING_ADDON_PACK to have a {STRING_ID} instead of concatenating strings. Again that might be better for localisation should languages want to change order.

**Comment by Gymnasiast** [comment]
_\*Sigh\*_
Couldn't you have asked this right away?

**Comment by IntelOrca** [comment]
@Gymnasiast no

**Comment by PFCKrutonium** [comment]
ITT: Sighs. (Sorry)

**Comment by Gymnasiast** [comment]
:/

Can you at least explain how to get the names to show up? I tried pushing them into RCT2_ADDRESS_COMMON_FORMAT_ARGS and several other things.

**Comment by IntelOrca** [comment]
```c
rct_string_id expansionNameId = w/e;
format_string(string_buffer, STR_REQUIRES_THE_FOLLOWING_ADDON_PACK, &expansionNameId);
```

**Comment by Gymnasiast** [comment]
Still gives an empty spot where the stringId should go.

**Comment by Gymnasiast** [comment]
I'll do the latter then.

**Comment by Gymnasiast** [comment]
Ok, that should be it. I opted for the former.

**Pull request merged**